PaillierTests/UnitTest1.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using PaillierExt;
+using System.Text;
+using System.Linq;
+using System.Security.Cryptography;
+
+namespace PaillierTests
+{
+    [TestClass]
+    public class UnitTest1
+    {
+        [TestMethod]
+        public void TestZero()
+        {
+            Paillier algorithm = new PaillierManaged();
+            algorithm.Padding = PaillierPaddingMode.LeadingZeros;
+
+            for (algorithm.KeySize = 384; algorithm.KeySize <= 544; algorithm.KeySize += 8)
+            {
+                Paillier encryptAlgorithm = new PaillierManaged();
+                encryptAlgorithm.FromXmlString(algorithm.ToXmlString(false));
+
+                Paillier decryptAlgorithm = new PaillierManaged();
+                decryptAlgorithm.FromXmlString(algorithm.ToXmlString(true));
+
+                var z = new BigInteger(0);
+
+                var z_enc = encryptAlgorithm.EncryptData(z.getBytes());
+                var z_dec = decryptAlgorithm.DecryptData(z_enc);
+
+                for(int i = 0; i < z_dec.Length; i++)
+                {
+                    Assert.AreEqual(z_dec[i], 0);
```

Comment:
[CollectionAssert](https://msdn.microsoft.com/en-us/library/ms243763(v=vs.120).aspx) might be more appropriate